clientconn.go

```diff
@@ -225,7 +225,7 @@ func Dial(target string, opts ...DialOption) (*ClientConn, error) {
 func DialContext(ctx context.Context, target string, opts ...DialOption) (conn *ClientConn, err error) {
 	// At the end of this method, we kick the channel out of idle, rather than
 	// waiting for the first rpc.
-	opts = append([]DialOption{withDefaultScheme("passthrough")}, opts...)
+	opts = append([]DialOption{withDefaultScheme("passthrough"), WithTargetResolutionEnabled()}, opts...)
```

Comment:
Why is `WithTargetResolutionEnabled` added by default? IIUC, we want to keep the default behaviour the same as before, i.e. don't resolve the target address on the client.

Reply:
Here is my understanding:
1 `grpc.Dial()` uses `passthrough` by default.
2. `passthrough` doesn't resolve target hostnames on the client.
3. `WithTargetResolutionEnabled` causes target hostnames to be resolved on the client.

From 1 & 2, `grpc.Dial()` doesn't resolve target hostnames on the client by default. Adding `WithTargetResolutionEnabled` seems to break this behaviour.

Reply:
Actually, using the `WithTargetResolutionEnabled` will just use the target resolver and not by pass resolution. So if someone uses `WithTargetResolutionEnabled` with `grpc.Dial` and default resolver, it will use passthrough and not do the resoltion, but if they use "dns" with `grpc.Dial` it will use dns resolver and do the resolution. This will preserve the behaviour.